core/optaplanner-constraint-streams/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/ComparisonIndexer.java

```diff
@@ -103,4 +104,18 @@ public boolean isEmpty() {
         return comparisonMap.isEmpty();
     }
 
+    private static final class KeyComparator<Key_ extends Comparable<Key_>> implements Comparator<Key_> {
+
+        @Override
+        public int compare(Key_ o1, Key_ o2) { // Exists so that the comparison operations can be more easily debugged.
+            return o1.compareTo(o2);
+        }
+
+    }
```

Comment:
This comparator is technically unnecessary, but it has two nice benefits:

- First, it allows us to easily debug the comparison operations by putting a breakpoint here, and not somewhere deep in the `TreeMap`.
- Second, this gets rid of the weird async profiler comparator behavior.

I did not find any performance impact of this.

Reply:
I don't mind this, but this is actually a natural comperator and I suspect over time users will want to have `Joiners.lessThan(X::getAge, Y::getAge, new MyOwnAgeComperator())`